pkgs/applications/editors/vim/plugins/update.py

```diff
@@ -1,181 +1,238 @@
-#!/usr/bin/env python
-
-# run with:
-# $ nix run .\#vimPluginsUpdater
-# format:
-# $ nix run nixpkgs#python3Packages.ruff -- update.py
-# type-check:
-# $ nix run nixpkgs#python3Packages.mypy -- update.py
-# linted:
-# $ nix run nixpkgs#python3Packages.flake8 -- --ignore E501,E265,E402 update.py
-
-# If you see `HTTP Error 429: too many requests` errors while running this
-# script, refer to:
-#
-# https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/vim.section.md#updating-plugins-in-nixpkgs-updating-plugins-in-nixpkgs
-#
-# (or the equivalent file /doc/languages-frameworks/vim.section.md
-# from Nixpkgs master tree).
-#
-
-import inspect
-import json
-import logging
+#!/usr/bin/env nix-shell
+#!nix-shell -i python3 -p "python3.withPackages(ps: with ps; [ nixpkgs-updaters-library ])"
+
+from __future__ import annotations
+
+import dataclasses
 import os
-import subprocess
-import textwrap
+import sys
+import typing as t
 from pathlib import Path
-from typing import List, Tuple
 
-log = logging.getLogger("vim-updater")
+from loguru import logger
 
-# Import plugin update library from maintainers/scripts/pluginupdate.py
-ROOT = Path(os.path.dirname(os.path.abspath(inspect.getfile(inspect.currentframe()))))
-import importlib
+from nupd import utils
+from nupd.base import ABCBase
+from nupd.cli import app
+from nupd.fetchers import nurl
+from nupd.fetchers.github import (
+    GHRepository,
+    github_fetch_graphql,
+    github_fetch_rest,
+)
+from nupd.inputs.csv import CsvInput
+from nupd.models import Entry, EntryInfo, ImplClasses, MiniEntry, NupdModel
 
-import pluginupdate
-from pluginupdate import PluginDesc, run_nix_expr
+if t.TYPE_CHECKING:
+    import collections.abc as c
 
-treesitter = importlib.import_module("nvim-treesitter.update")
+# Always assunme CWD is the root of nixpkgs, later will figure out how to use the CLI flag
+ROOT = Path.cwd() / "pkgs/applications/editors/vim/plugins"
```

Comment:
It should also be possible to get the running file's location, however that's only useful when run directly from the nixpkgs tree. If the updater is installed into the nix store, you'd still need a cli flag and/or current working dir.

Reply:
We can use `Path(__file__).parent.resolve()` to get the current file's directory.

If the newer `nix run` or `nix build` commands are used, then `__file__` would be an in-store path. I guess, since the script is intended to be run using its shebang, this is a non-issue.